runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -254,10 +257,12 @@ protected boolean fillReadQueue(final long maxGlobal,
                         }
                     }
                 }
-
-                // If we STILL don't have the data, now we need to do a hole
-                // fill.
-                if (currentEntry.getType() == DataType.EMPTY) {
+                //FIXME To be replaced by cleaner interface. This is just a patch till the real fix comes in.
+                // If we STILL don't have the data, it means the writer is either dead or too slow.
+                // We will now try to make sure we have committed data for the currentEntry. We will first
+                // try a HOLE FILL, failing which we will re-read the entry to see if the writer completed.
+                // We will try this in a loop till we get committed data for the currentEntry.
+                while (currentEntry.getType() == DataType.EMPTY) {
```

Comment:
This bug is a short term fix where the reads block till they read committed data. This is a requirement from the read interface. The larger interface changes will fix the problem you have mentioned in #439. 
The interface changes proposed are
1. Reads will surface only committed data.
2. Hole filling is an abstraction that sits somewhere between address space and replication, so it should be in the abstract replication layer. I think this is where we should fix the bug in #439.



It does fix one problem small minor problem in a chain:
1. If the actual write completes, read will finally read it.
2. If the hole fill fails for some reason, it will keep trying. Assuming that hole fill is the data that will be eventually put into the address space.